src/util/BinaryFuseFilter.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "lib/binaryfusefilter.h"
+#include "util/NonCopyable.h"
+#include "xdr/Stellar-ledger-entries.h"
+#include "xdr/Stellar-types.h"
+
+#include <cereal/archives/binary.hpp>
+#include <xdrpp/xdrpp/cereal.h>
+
+namespace stellar
+{
+
+// This class is a wrapper around the binary_fuse_t library that provides
+// serialization for the XDR BinaryFuseFilter type and provides a deterministic
+// LedgerKey interface.
+
+// Only allow uint8_t, uint16_t, and uint32_t
+template <typename T,
+          typename = std::enable_if_t<std::is_unsigned<T>::value &&
+                                      !std::is_same<T, uint64_t>::value>>
```

Comment:
`std::is_same<T,uint8_t> || std::is_same<T, uint16_t> || std::is_same<T,uint32_t>` and then you can remove the comment!